src/vmspawn/vmspawn.c

```diff
@@ -1257,6 +1319,7 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
         _cleanup_close_ int notify_sock_fd = -EBADF;
         _cleanup_strv_free_ char **cmdline = NULL;
         _cleanup_free_ int *pass_fds = NULL;
+        SSHInfo ssh_info;
```

Comment:
Maybe add a comment that this is only used conditionally, but must survive until the function finishes? I think it'd be better to even move this before definition of `bus`, to underline the fact that it needs to survive longer.